kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigConfigMap.java

```diff
@@ -306,19 +326,99 @@ private static GenericKubernetesResource filterResourceFromRef(VirtualKafkaClust
+            return Optional.of(new VirtualCluster(
+                    ResourcesUtil.name(cluster),
+                    new TargetCluster(bootstrap, buildTargetClusterTls(context, cluster)),
+                    null,
+                    Optional.empty(),
+                    virtualClusterIngressModel.gateways(),
+                    false, false,
+                    filterNamesForCluster(cluster)));
+        }
+        catch (InvalidClusterException e) {
+            SharedKafkaProxyContext.addClusterCondition(context, cluster, e.accepted());
+        }
+        return Optional.empty();
+    }
+
+    @NonNull
+    private static Optional<Tls> buildTargetClusterTls(Context<KafkaProxy> context,
+                                                       VirtualKafkaCluster cluster) {
+        return Optional.ofNullable(cluster.getSpec())
+                .map(VirtualKafkaClusterSpec::getTargetCluster)
+                .map(io.kroxylicious.kubernetes.api.v1alpha1.virtualkafkaclusterspec.TargetCluster::getTls)
+                .map(io.kroxylicious.kubernetes.api.v1alpha1.virtualkafkaclusterspec.targetcluster.Tls::getCertificateRef)
+                .map(certRef -> {
+                    return new Tls(new KeyPair(
+                            virtualClusterTargetTlsKeyPath(context, cluster, certRef).containerPath().toString(),
+                            virtualClusterTargetTlsCertPath(context, cluster, certRef).containerPath().toString(),
+                            null),
+                            null, null, null);
+                });
+    }
+
+    private static ContainerFileReference virtualClusterTargetTlsKeyPath(Context<KafkaProxy> context,
+                                                                         VirtualKafkaCluster cluster,
+                                                                         CertificateRef certRef) {
+        return virtualClusterTargetTlsPath(context, cluster, certRef, "tls.key");
+    }
+
+    private static ContainerFileReference virtualClusterTargetTlsCertPath(Context<KafkaProxy> context,
+                                                                          VirtualKafkaCluster cluster,
+                                                                          CertificateRef certRef) {
+        return virtualClusterTargetTlsPath(context, cluster, certRef, "tls.crt");
+    }
+
+    private static ContainerFileReference virtualClusterTargetTlsPath(Context<KafkaProxy> context,
+                                                                      VirtualKafkaCluster cluster,
+                                                                      CertificateRef certRef,
+                                                                      String file) {
+        if ("Secret".equals(certRef.getKind())
+                && certRef.getGroup().isEmpty()) {
+            // Assume it's compatible with a Secret of `type: kubernetes.io/tls`
```

Comment:
Not that the operator is not validating the secret (it's not even getting it). If it's not a tls cert the operator won't update the status to explain what's wrong.